### PR TITLE
Added scroll view to search page

### PR DESCRIPTION
**Pull request by bbenson09**
NOTE: There is still a bug in this! Search page correctly loads places sometimes, and other times it crashes. Thinking it is an issue with the timing of asynchronous block in getGMSPlaceFromPlace.

This branch adds a new method to the APIManager that queries a fixed number of places from Parse, rather than returning all the places, and returns them in the order that they were added to the database.  It also adds a scroll view to the search page so that the next ten places are queried from Parse when the user scrolls to the bottom of the page.

### CODE REVIEW

**Comment by bbenson09** [comment]
Bugs in scroll view are now fixed. Search page no longer crashes, infinite scroll loads data more than once, and scroll no longer jerks when loading more data.

File: APIs/APIManager.m [hunk 1]
```diff
@@ -107,9 +113,49 @@ - (void)getAllGMSPlaces:(void(^)(NSArray<GMSPlace*>* places))completion {
...
+            NSArray<Place *> *orderedObjects = [NSArray arrayWithArray:objects];
+            
+            Place *placeHolder = [[Place alloc] init];
+            NSMutableArray *places = [NSMutableArray arrayWithCapacity:kQuerySize];
+            for (int i = 0; i < orderedObjects.count; i++) {
+                [places addObject:placeHolder];
+            }
+            
+            __block NSNumber *count;
+            count = [NSNumber numberWithInteger:0];
```
Comment (stephinphection) [inline APIs/APIManager.m]: I believe you can do __block int count = 0;

using an int will just be a little simpler than an NSNumber as you don't have to keep reconstructing it and getting its intvalue

File: APIs/APIManager.m [hunk 2]
```diff
@@ -107,9 +113,49 @@ - (void)getAllGMSPlaces:(void(^)(NSArray<GMSPlace*>* places))completion {
...
+            for (int i = 0; i < orderedObjects.count; ++i) {
+                Place *myPlace = orderedObjects[i];
+                
+                // convert each Place to a GMSPlace
+                [self GMSPlaceFromPlace:myPlace withCompletion:^(GMSPlace *place) {
+                    [places replaceObjectAtIndex:i withObject:place];
+                    count = [NSNumber numberWithInt:[count intValue] + 1];
+                    if ([count isEqual:[NSNumber numberWithInteger:places.count]]) {
+                        [places removeObjectIdenticalTo:placeHolder];
```
Comment (stephinphection) [inline APIs/APIManager.m]: i believe this line should be unnecessary, once count gets big enough

File: APIs/APIManager.m.orig [hunk 3]
```diff
@@ -0,0 +1,158 @@
+//
+//  APIManager.m
+//  social-maps
```
Comment (stephinphection) [inline APIs/APIManager.m.orig]: I'm wondering why this file is in here? with the .orig extension? should we delete it?

**Pull request closed without merging**